mongoengine/dereference.py

```diff
@@ -220,7 +220,8 @@ def _attach_objects(self, items, depth=0, instance=None, name=None):
                     elif isinstance(v, (dict, SON)) and '_ref' in v:
                         data[k]._data[field_name] = self.object_map.get(v['_ref'].id, v)
                     elif isinstance(v, dict) and depth <= self.max_depth:
-                        data[k]._data[field_name] = self._attach_objects(v, depth, instance=instance, name=name)
+                        item_name = "%s.%s.%s" % (name, k, field_name)
```

Comment:
You should be use .format method in this case. https://www.python.org/dev/peps/pep-3101/